doc/reference/common-weaknesses/tx-rejection.rst

```diff
@@ -0,0 +1,16 @@
+Failure to test transaction rejection
+=====================================
+
+Any transaction submitted by off-chain code may be rejected, despite being perfectly valid. The wallet balancing the transaction may run out of funds, the wallet owner may simply decide not to sign the transaction, a competing transaction may spend one of the transaction’s inputs first, or the transaction may simply fail to be confirmed before its validity interval expires. Contracts need to be written to respond sensibly to this.
```

Comment:
This is a bit confusing. It's using "transaction submission" to mean "the transaction is sent to the plutus-apps transaction submission endpoint", which still has to pass through balancing etc. before it gets *actually* submitted. But everywhere else "transaction submission" means "actually submitting the transaction to the blockchain", so this reads very weirdly. Indeed, some of the conditions are things that can go wrong before real transaction submission, and some afterwards.